README.md

```diff
@@ -1,268 +1,174 @@
+       'the user to be new to the application, ' +
+       'but the User PID already exists.')
+   }
+
+   fakeUserDatabase[user.pid] = Object.assign({id: identity.pid}, identity.attributes)
+   return Object.assign({ levelOfAssurance: identity.levelOfAssurance }, fakeUserDatabase[identity.pid])
+ },
+
+// A callback that saves the unique request ID associated with the SAML messages
+// to the user's session.
+// This function is called after the Verify Service Provider has generated and
+// returned the AuthnRequest and associated RequestID.
+// The requestID should be saved in a secure manner, and such that it
+// corresponds to the user's current session and can be retrieved in order to validate
+// that SAML response that is returned from the IDP corresponds to the original AuthnRequest.
+function saveRequestId (requestId, request) {
+
+  // The following is an example that saves the requestId using the express-session middleware
+  // This would require express-session to be initialised with a secure secret e.g:
+
+  // const session = require('express-session')
+  //
+  // app.use(session({
+  //  secret: 'super secret secure token',
+  //  resave: true,
+  //  saveUninitialized: true
+  // }))
+
+  request.session.requestId = requestId
+},
+
+// A callback that returns the requestId that corresponds to the user's session.
+// This is used by the Verify Service Provider to ensure SAMLResponses received from IDPS
+// correspond to a the user's active session.
+function loadRequestId (request) {
+
+  // The following is an example that retrieves the request ID from the aforementioned
+  // express-session object.
+  return request.session.requestId
+},
+// Service Entity Id
+'http://your-service-entity-id'
+// This is only required when your Verify Service Provider is set up to be multi tenanted.
+// If it is provided, it is passed to the Verify Service Provider with each request, and
+// used to identify this service.
+
+// Saml Form Template Location
+'saml-form-template.njk',
+// This is an optional parameter which can be used to style the saml form
+// used to send the authn request to Verify.
+// This template should only be rendered if Javascript has been disabled in the user's browser.
+// If provided, the ssoLocation and samlRequest recieved from the Verify Service Provider
+// will be provided to the named template for rendering.
+// If this is not provided, passport-verify will render a default auto posting form
+// with the correct attributes.
+
+// Level of Assurance
+// LEVEL_1 or LEVEL2 depending on your service's requirements. Defaults to LEVEL_2.
+'LEVEL_2'
+))
+```
+
+If your service connects to GOV.UK Verify using a Matching Service Adapter, you should use the `createStrategy` method to create a strategy when configuring passport.js.
+
+See the [API documentation for the createStrategy method](https://alphagov.github.io/passport-verify/modules/_passport_verify_strategy_.html#createstrategy) for more details.
```

Comment:
I wonder if this should be part of the previous paragraph, because it only applies to the people with MSA.

Reply:
Yes, that makes sense. I had it separate to try and improve readability because it seemed like a really long paragraph. But it's not that long when it's published. Only the markdown is miles long because of the link...